locations/spiders/edwardjones.py

```diff
@@ -1,39 +1,45 @@
-import scrapy
+import requests
+from scrapy.spiders import SitemapSpider
 
-from locations.linked_data_parser import LinkedDataParser
+from locations.hours import OpeningHours
+from locations.structured_data_spider import StructuredDataSpider
 
 
-class EdwardJonesSpider(scrapy.Spider):
+class EdwardJonesSpider(SitemapSpider, StructuredDataSpider):
     name = "edwardjones"
     item_attributes = {"brand": "Edward Jones", "brand_wikidata": "Q5343830"}
     allowed_domains = ["www.edwardjones.com"]
+    sitemap_urls = ["https://www.edwardjones.com/us-en/sitemap/financial-advisor/sitemap.xml"]
 
-    def get_page(self, n):
-        return scrapy.Request(
-            f"https://www.edwardjones.com/api/v2/financial-advisor/results?q=57717&distance=75000&distance_unit=mi&page={n}",
-            meta={"page": n},
-        )
-
-    def start_requests(self):
-        yield self.get_page(1)
-
-    def parse(self, response):
-        data = response.json()["results"]
-        if data == []:
-            return
-        for row in data:
-            url = "https://www.edwardjones.com" + row["faUrl"]
-            meta = {
-                "lat": row["lat"],
-                "lon": row["lon"],
+    def post_process_item(self, item, response, ld_data, **kwargs):
+        oh = OpeningHours()
+        oh.from_linked_data(ld_data, time_format="%H:%M:%S")
+        item["opening_hours"] = oh.as_opening_hours()
+
+        flag = True
+        page = 1
+        while flag:
+            url = "https://www.edwardjones.com/api/v3/financial-advisor/results?q={postcode}&distance=75&distance_unit=mi&page={page}&searchtype=2".format(
+                postcode=item["postcode"], page=page
+            )
+
+            headers = {
+                "Host": "www.edwardjones.com",
+                "Accept": "application/json",
+                "Referer": "https://www.edwardjones.com/us-en/search/financial-advisor/results",
             }
-            yield scrapy.Request(url, callback=self.parse_location, meta=meta)
-        yield self.get_page(1 + response.meta["page"])
-
-    def parse_location(self, response):
-        ld = LinkedDataParser.find_linked_data(response, "LocalBusiness")
-        del ld["openingHoursSpecification"]
-        item = LinkedDataParser.parse_ld(ld)
-        item["lat"] = response.meta["lat"]
-        item["lon"] = response.meta["lon"]
+            response = requests.get(url, headers=headers, data={}).json()
```

Comment:
Please use `yield` to generate a new request rather than using `requests` to  make HTTP calls inside the parse function.

Reply:
Updated the code. Thanks for the feedback.